### ISSUE TITLE
Problem parsing HTML with non standard attribute 

### ISSUE THREAD

**Issue by petemcfarlane** [deanblackborough/php-quill-renderer#125]
Hi,

We have some custom attributes to support tables in quill, see example delta:
```
{
  "ops": [
    {
      "insert": "world",
      "attributes": {
        "who": "3",
        "bold": true
      }
    },
    {
      "insert": "\n",
      "attributes": {
        "who": "3",
        "table-cell": {
          "id": "table-id-hxrct",
          "rowId": "table-row-efeap",
          "cellId": "table-cell-vjhos"
        }
      }
    },
    {
      "insert": "\n"
    }
  ]
}
```

My first problem is this code:
https://github.com/deanblackborough/php-quill-renderer/blob/07a7d0cd4e86c7f590787f52a01d61f95d04fbf4/src/Delta/Html/Compound.php#L128-L130
will try to interpolate the `$value` array as a string... which is causing an error. This could be easily fixed by just changing the `else` to `else if (is_string($value))`.

My second issue is I'd like to filter out some attributes, e.g. the "who" attribute. I'd rather this was not visible in the output at all. I can't currently see a way to handle custom attributes, by either silently ignoring them or adding custom attributes (apart from extending and just overriding the parsers). Do you have any suggestions or guidance for doing so?

Thanks

**Comment by deanblackborough** [comment]
Hi Pete

Yep, that else is pretty pointless, I'll push an update on Monday that will check for an array|string etc

I'm starting work on version 4.0 soon which will support plugins making cases like yours easy to handle.

Until then there is a simple change I can make, when you instantiate the renderer I can allow an ignore attributes array to be set. I'll try and roll that into the update on Monday as well.

Regards
Dean

**Comment by deanblackborough** [comment]
@petemcfarlane 

I've just pushed v3.18.0, two changes in the release.

- Anywhere that adds custom attributes will now ignore attributes and values that aren't strings.
- I have added the ability to define custom attributes which you would like the parser to ignore. 

If you check the changes for the newest release, the second test I added shows how to set attributes which the parser should ignore, in your case 'who' - https://github.com/deanblackborough/php-quill-renderer/commit/1eb2fb51d62addf9f4a048050a535197015987e8

Very soon I will start work on v4, the main feature being custom deltas via plugins; you will easily be able to define how deltas are rendered and also choose which deltas the parser should load. At this time I can't give a date on when v4 will be ready other than I need it for a project I'm working on so it will be soon.
